src/main/java/chatbot/impl/MessageParserImpl.java

```diff
@@ -0,0 +1,215 @@
+package chatbot.impl;
+
+import java.util.List;
+
+import chatbot.MessageParser;
+import chatbot.Task;
+import chatbot.TaskStorage;
+import chatbot.exceptions.InvalidMessageException;
+import chatbot.impl.tasks.DeadlineTask;
+import chatbot.impl.tasks.EventTask;
+import chatbot.impl.tasks.TodoTask;
+
+/**
+ * Implements the MessageParser interface to handle various types of user messages.
+ */
+public class MessageParserImpl implements MessageParser {
+
+    private final TaskStorage storage;
+
+    /**
+     * Constructs a MessageParserImpl with the specified task storage.
+     *
+     * @param storage The TaskStorage object for managing tasks.
+     */
+    public MessageParserImpl(TaskStorage storage) {
+        this.storage = storage;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation processes various commands including list, mark, unmark,
+     * todo, deadline, event, and delete. It throws an InvalidMessageException for
+     * unrecognized commands or invalid inputs.
+     */
+    @Override
+    public String handleMessage(String input) throws InvalidMessageException {
+        if (input.isEmpty()) {
+            throw new InvalidMessageException("Sorry, your message cannot be empty. :(");
+        }
+
+        String[] inputParts = input.split(" ", 2);
+        String command = inputParts[0];
+
+        return switch (command) {
+            case "list" -> handleList();
+            case "mark" -> handleMark(inputParts);
+            case "unmark" -> handleUnmark(inputParts);
+            case "todo" -> handleTodo(inputParts);
+            case "deadline" -> handleDeadline(inputParts);
+            case "event" -> handleEvent(inputParts);
+            case "delete" -> handleDelete(inputParts);
+            case "find" -> handleFind(inputParts);
+            default -> throw new InvalidMessageException("Sorry, I don't recognize that command. :(");
+        };
+    }
+
+    /**
+     * Handles the 'list' command.
+     *
```

Comment:
Although JavaDoc for private methods was not mentioned in requirements, it's great that you've done it! Improves readability a lot